Draft out first chapter of tutorial

klappradla opened the pull request:
As discussed, @nickgnd and me put together a rough outline of "chapters" for the tutorial and already added a draft text for the first chapter.

**NOTE:** we wrote it as basically one sentence per line as we initially thought this is easier for feedback and diffing... but feel free to disagree and I move it into proper formatting.

nickgnd commented on GUIDES.md:
```diff
@@ -73,7 +73,117 @@ The playing field is a grid of tiles, addressable like a coordinate system.
+### 10. Potential later steps
+
+- move things into components (e.g. the score, the snake)
+- add multiplayer
```
Ops...we forgot to delete this file since we moved it inside the `docs` folder 🤦‍♂

klappradla replied:
No. Reach for the stars 🌠

ggpasqualino replied:
I think he means this content was moved to `docs/GUIDES.md` but it's duplicated here

klappradla replied:
Oh hell 🙈 

Thanx!

ggpasqualino commented on docs/GUIDES.md:
```diff
@@ -73,7 +73,118 @@ The playing field is a grid of tiles, addressable like a coordinate system.
+
+### 0. Prerequisites
+
+- explain what Scenic is
```
I was thinking, maybe the elixir section above should be here...

nickgnd replied:
If I remember well from the last meeting, the elixir section should be moved in a separated guide.

https://github.com/codecurious-bln/elixir-orga-team/blob/master/meetings/2019-10-29.md#outcome

ggpasqualino commented on docs/GUIDES.md:
```diff
@@ -73,7 +73,118 @@ The playing field is a grid of tiles, addressable like a coordinate system.
+![sceenshot](images/01-scenic-new-screen.png)
+
+
+### 2. Draw a worm
```
and the grid section should be under this section because it's already part of the game solution itself

nickgnd replied:
Totally 👍 , and maybe adding a prior chapter for introducing the grid concept without the worm 🤔

klappradla replied:
👍 for keeping the concepts closer to their usage.

I moved the grid explanation down into the respective topic (did not formulate it out though).

ggpasqualino commented on docs/GUIDES.md:
```diff
@@ -73,7 +73,118 @@ The playing field is a grid of tiles, addressable like a coordinate system.
+## Tutorial
+
+### 0. Prerequisites
+
```
it may sound silly, but I think we should have a `explain what snake is`

nickgnd replied:
Good point :+1: 
I'm not sure if this part belongs to the `Prerequisites` section, I would  rather add it to an introduction part at the beginning which explains:
- what it is the scope of the game
- How it works
- etc.

What do you think?

Maybe we can also find a better name than `prerequisites` 🤔 like: `Before starting` or similar

nickgnd replied:
Should we create an issue for tracking that? 
This "introduction" part is not strictly related to the first chapter, so I would not tackle it right now but in another PR.

klappradla replied:
👍 to everything said here 🙇 

Outlining the "goal" or "product" of the workshop is something I'm also missing from the [rails girls tutorial](http://guides.railsgirls.com/app). We should definitely have that (and it would be the place to explain what snake is).

I would definitely want to a picture like this then as well 🎉 
![image](https://user-images.githubusercontent.com/3491815/68298925-e24ecc80-009a-11ea-89cb-6ea0b594e2a4.png)

I initially thought we'd have a *"concepts"* section in the beginning for explaining everything. But reading your comments, I'm changing my mind: it's too dry and too much unnecessary information for getting started with the tutorial itself. Maybe it makes more sense, to spread this conceptual information more though the tutorial and put it closer to where it applies.

So I'd suggest to have an *"About"* section in the beginning, where we give a rough outline of what we will be doing (including what snake is). *"Prerequisites*" would then in my opinion still be a separate next section to prepare for the actual hands on coding.

What do you think?

ggpasqualino replied:
Sounds like a good approach to me

![agreed](https://media.giphy.com/media/iU8395XCj6ltC/giphy.gif)

klappradla replied:
Ok, moved this into https://github.com/codecurious-bln/elixir_snake/issues/10 and will tackle the section in a separate PR.

Fire for you?

ggpasqualino commented on docs/GUIDES.md:
```diff
@@ -73,7 +73,118 @@ The playing field is a grid of tiles, addressable like a coordinate system.
+
+The task makes some assumptions about the typical structure of a Scenic application.
+It will generate a skeleton for our snake app with all directories and files already in place.
+This is "boilderplate" code we would otherwise need to write by hand.
```
```suggestion
This is "boilerplate" code we would otherwise need to write by hand.
```

ggpasqualino commented on docs/GUIDES.md:
```diff
@@ -73,7 +73,118 @@ The playing field is a grid of tiles, addressable like a coordinate system.
+    You can also run it interactively like this:
+        $ iex -S mix
+
+We see it generated some files inside a new `snake` folder and gives us hints what we can do next.
```
```suggestion
We see it generated some files inside a new `snake` folder and gives us hints about what we can do next.
```

ggpasqualino commented on docs/GUIDES.md:
```diff
@@ -73,7 +73,118 @@ The playing field is a grid of tiles, addressable like a coordinate system.
+
+We see it generated some files inside a new `snake` folder and gives us hints what we can do next.
+
+Fist, we need to Navigate into our new project directory:
```
```suggestion
First, we need to Navigate into our new project directory:
```

klappradla commented:
I think™ I addressed all the comments - maybe you want to quickly double check again @nickgnd & @ggpasqualino ?

Most importantly: I moved the grid explanation into the respective section and opened an issue for the introduction section that I'd then tackle in a separate PR. I already have a few ideas for it, but open for more.
